editor/src/clj/editor/pipeline/texture_set_gen.clj

```diff
@@ -77,38 +76,73 @@
+  (let [points (vector-of :float -0.5 -0.5 -0.5 0.5 0.5 0.5 0.5 -0.5)
+        uvs (vector-of :float 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0)
+        indices (vector-of :int 0 1 2 0 2 3)]
+    (-> (TextureSetProto$SpriteGeometry/newBuilder)
+        (.addAllVertices points)
+        (.addAllUvs uvs)
+        (.addAllIndices indices)
+        (.buildPartial))))
+
+(defn- make-rect-sprite-geometry
+  ^TextureSetProto$SpriteGeometry [^long width ^long height]
+  (-> (TextureSetProto$SpriteGeometry/newBuilder rect-sprite-geometry-template)
+      (.setWidth width)
+      (.setHeight height)
+      (.build)))
+
+(defn- make-image-sprite-geometry
+  ^TextureSetProto$SpriteGeometry [^Image image]
+  (let [error-node-id (:digest-ignored/error-node-id image)
+        resource (.path image)
+        sprite-trim-mode (.sprite-trim-mode image)]
+    (assert (g/node-id? error-node-id))
+    (assert (resource/resource? resource))
+    (case sprite-trim-mode
+      :sprite-trim-mode-off (make-rect-sprite-geometry (.width image) (.height image))
+      (let [buffered-image (resource-io/with-error-translation resource error-node-id :image
+                             (image-util/read-image resource))]
+        (g/precluding-errors buffered-image
+          (let [hull-vertex-count (sprite-trim-mode->hull-vertex-count sprite-trim-mode)]
+            (TextureSetGenerator/buildConvexHull buffered-image hull-vertex-count)))))))
+
 (defn atlas->texture-set-data
-  [animations images margin inner-padding extrude-borders workspace]
-  (let [img-to-index (into {} (map-indexed #(vector %2 (Integer. ^int %1)) images))
-        anims-atom (atom animations)
-        anim-imgs-atom (atom [])
-        anim-iterator (reify TextureSetGenerator$AnimIterator
-                        (nextAnim [this]
-                          (let [anim (first @anims-atom)]
-                            (reset! anim-imgs-atom (or (:images anim) []))
-                            (swap! anims-atom rest)
-                            (anim->AnimDesc anim)))
-                        (nextFrameIndex [this]
-                          (let [img (first @anim-imgs-atom)]
-                            (swap! anim-imgs-atom rest)
-                            (img-to-index img)))
-                        (rewind [this]
-                          (reset! anims-atom animations)
-                          (reset! anim-imgs-atom [])))
-        rects (map map->Rect images)
-        sprite-geometries (map (fn [{:keys [path sprite-trim-mode] :as _image}]
-                                 (let [image-resource (workspace/find-resource workspace path)
-                                       buffered-image (image-util/read-image image-resource)
-                                       hull-vertex-count (sprite-trim-mode->hull-vertex-count sprite-trim-mode)]
-                                   (TextureSetGenerator/buildConvexHull buffered-image hull-vertex-count)))
```

Comment:
All of this is mostly indentation due to generating the `sprite-geometries` first and returning early if there are errors reading the images.